homeassistant/components/template/event.py

```diff
@@ -0,0 +1,238 @@
+"""Support for events which integrates with other components."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any, Final
+
+import voluptuous as vol
+
+from homeassistant.components.event import (
+    DOMAIN as EVENT_DOMAIN,
+    ENTITY_ID_FORMAT,
+    EventDeviceClass,
+    EventEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_DEVICE_CLASS
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity_platform import (
+    AddConfigEntryEntitiesCallback,
+    AddEntitiesCallback,
+)
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from . import TriggerUpdateCoordinator
+from .entity import AbstractTemplateEntity
+from .helpers import (
+    async_setup_template_entry,
+    async_setup_template_platform,
+    async_setup_template_preview,
+)
+from .template_entity import (
+    TEMPLATE_ENTITY_COMMON_CONFIG_ENTRY_SCHEMA,
+    TemplateEntity,
+    make_template_entity_common_modern_attributes_schema,
+)
+from .trigger_entity import TriggerEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Template Event"
+
+CONF_EVENT_TYPE = "event_type"
+CONF_EVENT_TYPES = "event_types"
+
+DEVICE_CLASS_SCHEMA: Final = vol.All(vol.Lower, vol.Coerce(EventDeviceClass))
+
+EVENT_COMMON_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASS_SCHEMA,
+        vol.Required(CONF_EVENT_TYPE): cv.template,
+        vol.Required(CONF_EVENT_TYPES): cv.template,
```

Comment:
Why should this be a template instead of a fixed list?
Note: I'm not saying it shouldn't, but I'm asking for a motivation. FWIW, the select template platform's options are defined by a template which is conceptually somewhat similar.